pkg/util/ciutil/detect.go

```diff
@@ -16,88 +16,130 @@ package ciutil
 
 import (
 	"os"
+
+	"github.com/pulumi/pulumi/pkg/util/ciutil/systems"
 )
 
 // detectors contains environment variable names and their values, if applicable, for detecting when we're running in
 // CI. See https://github.com/watson/ci-info/blob/master/index.js
-var detectors = map[System]detector{
-	AppVeyor:                    envVarDetector{envVar: []string{"APPVEYOR"}},
-	AWSCodeBuild:                envVarDetector{envVar: []string{"CODEBUILD_BUILD_ARN"}},
-	AtlassianBamboo:             envVarDetector{envVar: []string{"bamboo_planKey"}},
-	AtlassianBitbucketPipelines: envVarDetector{envVar: []string{"BITBUCKET_COMMIT"}},
-	AzurePipelines:              envVarDetector{envVar: []string{"TF_BUILD"}},
-	Buildkite:                   envVarDetector{envVar: []string{"BUILDKITE"}},
-	CircleCI:                    envVarDetector{envVar: []string{"CIRCLECI"}},
-	Codeship:                    envValueDetector{envMap: map[string]string{"CI_NAME": "codeship"}},
-	Drone:                       envVarDetector{envVar: []string{"DRONE"}},
-	GenericCI:                   envVarDetector{envVar: []string{"GENERIC_CI_SYSTEM"}},
-	GitHub:                      envVarDetector{envVar: []string{"GITHUB_WORKFLOW"}},
-	GitLab:                      envVarDetector{envVar: []string{"GITLAB_CI"}},
-	GoCD:                        envVarDetector{envVar: []string{"GO_PIPELINE_LABEL"}},
-	Hudson:                      envVarDetector{envVar: []string{"HUDSON_URL"}},
-	Jenkins:                     envVarDetector{envVar: []string{"JENKINS_URL", "BUILD_ID"}},
-	MagnumCI:                    envVarDetector{envVar: []string{"MAGNUM"}},
-	Semaphore:                   envVarDetector{envVar: []string{"SEMAPHORE"}},
-	TaskCluster:                 envVarDetector{envVar: []string{"TASK_ID", "RUN_ID"}},
-	TeamCity:                    envVarDetector{envVar: []string{"TEAMCITY_VERSION"}},
-	Travis:                      envVarDetector{envVar: []string{"TRAVIS"}},
-}
+var detectors = map[systems.SystemName]systems.System{
```

Comment:
Not sure how the "See watson/ci-info" link is relevant. Perhaps remove it?

Reply:
Looking at https://github.com/watson/ci-info/blob/master/index.js it seems that it may have been used to indicate the source of inspiration for the type of checks that we do here. That is, using specific env vars to detect if we are running inside a specific CI environment.